### PR TITLE
fix: pass uds socket flag to peers

### PR DESCRIPTION
**Pull request by dranikpg**
Andy if you read this, please be wary of rebasing onto this once merged because it'll break Dragonfly

Needed for https://github.com/dragonflydb/dragonfly/pull/2033

### CODE REVIEW

File: util/fibers/uring_socket.cc [hunk 1]
```diff
@@ -77,7 +77,7 @@ auto UringSocket::Accept() -> AcceptResult {
     int res = accept4(real_fd, NULL, NULL, SOCK_NONBLOCK | SOCK_CLOEXEC);
     if (res >= 0) {
       UringSocket* fs = new UringSocket{nullptr};
-      fs->fd_ = res << kFdShift;
+      fs->fd_ = (res << kFdShift) | (fd_ & kInheritedFlags);
```
Comment (dranikpg) [inline util/fibers/uring_socket.cc:80]: Because we didn't actually pass this flag, IsUDS() was false for all uds peers

Comment (romange) [reply util/fibers/uring_socket.cc:80]: I think there is fiber_socket_test or something. Can we cover it there?

**Comment by dranikpg** [comment]
Done. It was the most painful test I've written so far 🙂  (😭)

File: util/fibers/fiber_socket_base.cc [hunk 2]
```diff
@@ -193,7 +193,7 @@ auto LinuxSocketBase::LocalEndpoint() const -> endpoint_type {
   if (fd_ < 0)
     return endpoint;
 
-  DCHECK_EQ(0, fd_ & IS_UDS);
+  CHECK_EQ(0, fd_ & IS_UDS);
```
Comment (romange) [inline util/fibers/fiber_socket_base.cc:196]: for sake of avoiding stupid crashes for no reason during codepaths that are not covered by our tests I suggest returning an empty endpoint like with fd_ < 0

**Comment by romange** [comment]
thanks for writing the test :) I did not remember that we do not have any client side code for UDS 🤷🏼

File: util/fibers/fiber_socket_base.cc (hunk 2 above)
Comment (dranikpg) [reply util/fibers/fiber_socket_base.cc]: Sure, did it for RemoteEndpoint() as well. Yet I was planning to review all usages of Endpoint() within Dragonfly as we call them for UDS as well

**Review by romange — APPROVED** [review]

**Pull request merged**